### PR TITLE
test: add integration tests for `Pausable` plugins

### PR DESCRIPTION
**Pull request by mooori**
Previously `Pausable` was tested by unit tests via [`near_sdk::testing_env`](https://docs.rs/near-sdk/4.1.1/near_sdk/macro.testing_env.html), which mocks the chain.

The new integration tests compile the contract in `near-plugins/tests/contracts/pausable`, deploy it on chain via near `workspaces` and send transactions to it.

This is a step towards having _only one_ contract for the `Pausable` plugin which is used for tests and serves as documentation resp. example, as laid out in #48. In a follow up PR I’ll clean up duplicate example contracts.

## Other changes: refactor `near-plugins/tests/common`
- Removed superfluous types that complicated the code.
- Generalized helper functions to be used in tests of multiple plugins. Prior only `AccessControllable` was tested via integration tests.

### CODE REVIEW

File: near-plugins/tests/contracts/pausable/Makefile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+build:
```
Comment (birchmd) [inline near-plugins/tests/contracts/pausable/Makefile:1]: I think `build` and `expand` should be added as ["phony" targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html) to following Makefile conventions.

File: near-plugins/tests/contracts/pausable/rust-toolchain [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[toolchain]
+channel = "1.64.0"
```
Comment (birchmd) [inline near-plugins/tests/contracts/pausable/rust-toolchain:2]: Could probably use 1.65.0 if we wanted to.

File: near-plugins/tests/pausable.rs [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+    assert_insufficient_acl_permissions(
+        result,
+        "pa_pause_feature",
+        vec!["PauseManager".to_string()],
+    );
+}
+
+#[tokio::test]
+/// Only accounts that were granted a manager role may pause features.
+/// TODO same for unpause
```
Comment (birchmd) [inline near-plugins/tests/pausable.rs:177]: Remaining TODO item

**Review by birchmd — COMMENTED** [review]
Looks pretty good to me, nice job! Just a few nits.

File: near-plugins/tests/pausable.rs (hunk 3 above)
Comment (mooori) [reply near-plugins/tests/pausable.rs]: Sorry - done in 1b69596.

File: near-plugins/tests/contracts/pausable/Makefile (hunk 1 above)
Comment (mooori) [reply near-plugins/tests/contracts/pausable/Makefile:1]: Done in 93887f3.

File: near-plugins/tests/contracts/pausable/rust-toolchain (hunk 2 above)
Comment (mooori) [reply near-plugins/tests/contracts/pausable/rust-toolchain:2]: Updating to `1.65.0` works fine locally on my machine, however it makes [CI fail](https://github.com/aurora-is-near/near-plugins/actions/runs/3747724193/jobs/6364726907). Perhaps CI cannot handle differing channels in [root](https://github.com/aurora-is-near/near-plugins/blob/b679c29fb45698d53cfed8abcc37a64664c57907/rust-toolchain) and test contract toolchains.

I openend #52 and would leave the channel as is for this PR.

File: near-plugins/tests/contracts/pausable/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                "Method paused for large values of counter"
+            );
+        }
+
+        self.counter += 1;
+    }
+}
+
+/// Exposing internal methods to facilitate integration testing.
+/// TODO needed for this contract?
```
Comment (birchmd) [inline near-plugins/tests/contracts/pausable/src/lib.rs:118]: Is this question answered now?

**Review by birchmd — APPROVED** [review]
LTGM, nice work

File: near-plugins/tests/contracts/pausable/src/lib.rs (hunk 4 above)
Comment (mooori) [reply near-plugins/tests/contracts/pausable/src/lib.rs]: It turned out that these functions are not needed for the `Pausable` test contract. They're removed in 5fe4f37.

Thanks for catching another open TODO!

**Comment by mooori** [comment]
CI is green and I've double checked that there are no more open TODOs related to this PR, so merging now.

**Review by mrLSD — APPROVED** [review]

**Pull request merged**